sycl/plugins/rocm/pi_rocm.cpp

```diff
@@ -0,0 +1,4794 @@
+//==---------- pi_rocm.cpp - HIP Plugin -----------------------------------==//
```

Comment:
This file contains many commented source lines. Please, consider cleanup. I believe, whatever issues there are, they can be converted to GitHub issues.

Reply:
At present, the support of rocm is not as perfect as cuda, and some runtime APIs are not supported, such as `hipLinkComplete`,` hipMemsetD2D32Async`, and so on. We are considering whether there are other solutions, or temporarily remove these functions?

Reply:
Yes, I think it makes sense to replace non-working parts with an assertion (just like Level Zero and CUDA plugins do). Instead you can create GitHub issues for PI APIs, that are not fully supported. It will be a great place to start for people who want to contribute to AMD support in DPC++.